statescript/statescript_test.go

```diff
@@ -190,6 +191,36 @@ func TestExecutor(t *testing.T) {
 	sysInstallScripts, _, err = e.get("ArtifactInstall", "Leave")
 	testArtifactArrayEquals(t, scriptArr[1:], sysInstallScripts)
 	assert.NoError(t, err)
+
+	// add a script that will time-out, and die
+	filep, err := createArtifactTestScript(tmpArt, "ArtifactInstall_Leave_10_btoot", "!#/bin/bash \nsleep 2")
+	assert.NoError(t, err)
+	ret := execute(filep.Name(), 1)
+	assert.Equal(t, ret, -1)
+
+	// Test retry-later functionality
+	l := Launcher{
+		ArtScriptsPath:          tmpArt,
+		RootfsScriptsPath:       tmpRootfs,
+		SupportedScriptVersions: []int{2, 3},
+		RetryInterval:           1,
+		RetryTimeout:            2,
+	}
+
+	// add a script that will time out
+	_, err = createArtifactTestScript(tmpArt, "ArtifactInstall_Enter_66", "#!/bin/bash \n sleep 1\n exit 21")
+	assert.NoError(t, err)
+	err = l.ExecuteAll("ArtifactInstall", "Enter", false)
+	assert.Contains(t, err.Error(), "retry time-limit exceeded")
+
+	err = os.Remove(filepath.Join(tmpArt, "ArtifactInstall_Enter_66"))
+	assert.NoError(t, err)
+
+	// add a script that retries and then succeeds
+	script := fmt.Sprintf("#!/bin/bash \n sleep 1 \n if [ ! -f %s/scriptflag ]; then\n echo f > %[1]s/scriptflag\n exit 21 \nfi \n rm -f %[1]s/scriptflag\n exit 0", tmpArt)
+	_, err = createArtifactTestScript(tmpArt, "ArtifactInstall_Enter_67", script)
```

Comment:
I think this one is not removed as well, unless we are removing the whole `tmpArt` directory when the tests are over.

Reply:
`tmpArt` is removed after the tests are run ;)